shard_consumer.go

```diff
@@ -46,6 +50,8 @@ func getShardIterator(k kinesisiface.KinesisAPI, streamName string, shardID stri
 		ShardIteratorType:      &shardIteratorType,
 		StartingSequenceNumber: ps,
 		StreamName:             aws.String(streamName),
+		Timestamp:              iteratorStartTimestamp,
+		// TODO: verify that providing this argument as explicitly nil doesn't affect other iterator types.
```

Comment:
Not setting it would mean it's `nil` so it shouldn't be any different if `iteratorStartTimestamp` is `nil`.

Reply:
Ah, I tested this and confirmed that you're right. Forgot to remove the comment. :)